### DISCUSSION TITLE
图题想要实现“单行居中，多行左对齐”

### DISCUSSION THREAD

**Discussion by GodfatherPacino** [sjtug/SJTUThesis#785] [提问] [answered]
您好，
我的论文中因为图题caption比较长，很多图都出现了中英文图题多行的情况。
预答辩的时候，老师要求我，多行图题就跟正文一样左对齐，单行的时候居中。
目前的模板中，单行和多行都是默认居中。
我通过自己尝试将sjtuthesis.cls中修改为：`\captionsetup{%
  format        = plain,
%   format        = hang,
  labelformat   = simple,
  labelsep      = enskip,
%   justification = centering,
  justification = raggedright,
  singlelinecheck = on,
  font          = sjtu@caption@font
}`
但是目前是多行左对齐了，单行也左对齐了，`singlelinecheck = on`并没有发挥判断单行居中的作用。
试了好多了，头疼，求助。

**Comment by GodfatherPacino** [comment]
补充一下，我发现目前单行中文不居中，是因为英文是多行。如何解决

**Reply by GodfatherPacino** [reply]
我把sjtuthesis.cls中原来的`\captionsetup[bi-second]{bi-second}`改成`\captionsetup[bi-second]{justification=raggedright}`，并不行。

**Comment by AlexaraWu** [comment, marked as answer]
~~`justification = centerlast`~~

**Update:** 

抱歉，之前没有认真审题 😂 

```latex
\captionsetup{%
  format        = plain,
  labelformat   = simple,
  labelsep      = enskip,
  justification = justified,
  bi-slc        = off,
  font          = sjtu@caption@font
}
```

![image](https://user-images.githubusercontent.com/15205102/162619418-d1df6e37-1588-4a95-8620-7e00ff2cec67.png)

**Reply by GodfatherPacino** [reply]
<img width="321" alt="Snipaste_2022-04-10_20-20-31" src="https://user-images.githubusercontent.com/33151790/162617822-08fa8151-13a2-4760-ac64-984db39a6b75.png">
刚试了一下，变成了这个效果。在多行情况下也不左对齐了。。。

**Reply by GodfatherPacino** [reply]
代码如下：`\captionsetup{%
  format        = plain,
%   format        = hang,
  labelformat   = simple,
  labelsep      = enskip,
%   justification = centering,
%   justification = raggedright,
  justification = centerlast,
  singlelinecheck = on,
  font          = sjtu@caption@font
}
\captionsetup[sub]{%
  format        = hang,
  labelformat   = brace,
  justification = justified,
  font          = sjtu@subcaption@font
}
\DeclareCaptionOption{bi-second}[]{%
  \def\tablename{\sjtu@name@table@second}
  \def\figurename{\sjtu@name@figure@second}
}
\captionsetup[bi-second]{bi-second}`

**Reply by AlexaraWu** [reply]
回答已更新，之前没有仔细看需求。有一个针对 `bicaption` 的 Single line check 选项，关闭它就可以了。

**Reply by GodfatherPacino** [reply]
非常感谢，成功了！
我帮你把逗号补上:)

> \captionsetup{%
  format        = plain,
  labelformat   = simple,
  labelsep      = enskip,
  justification = justified,
  bi-slc        = off,
  font          = sjtu@caption@font
}
